Tests/L2Tests/L2TestsPlugin/tests/Telemetry_L2Test.cpp

```diff
@@ -0,0 +1,484 @@
+    EXPECT_FALSE(result.HasLabel("errorCode"));
+    EXPECT_FALSE(result.HasLabel("errorDescription"));
+
+    /* Register for temperature threshold change event. */
+    status = jsonrpc.Subscribe<JsonObject>(JSON_TIMEOUT,
+                                           _T("onReportUpload"),
+                                           &AsyncHandlerMock_Telemetry::onReportUpload,
+                                           &async_handler);
+
+    EXPECT_EQ(Core::ERROR_NONE, status);
+
+    message = "{\"telemetryUploadStatus\":\"UPLOAD_SUCCESS\"}";
+    expected_status.FromString(message);
+    EXPECT_CALL(async_handler, onReportUpload(MatchRequestStatus(expected_status)))
+        .WillOnce(Invoke(this, &Telemetry_L2test::onReportUpload));
+
+    //params["telemetryUploadStatus"] = "UPLOAD_SUCCESS";
+    status = InvokeServiceMethod("org.rdk.Telemetry.1", "uploadReport", params, result);
+    EXPECT_EQ(Core::ERROR_NONE, status);
+    EXPECT_FALSE(result["success"].Boolean());
+    EXPECT_STREQ("null", result["value"].String().c_str());
+
+
+    ON_CALL(*p_rBusApiImplMock, rbus_open(::testing::_, ::testing::_))
+        .WillByDefault(
+            ::testing::Return(RBUS_ERROR_SUCCESS));
+
+    EXPECT_CALL(*p_rBusApiImplMock, rbusMethod_InvokeAsync(::testing::_, ::testing::_, ::testing::_, ::testing::_, ::testing::_))
+        .Times(2)
+        .WillOnce(::testing::Invoke(
+            [&](rbusHandle_t handle, char const* methodName, rbusObject_t inParams, rbusMethodAsyncRespHandler_t callback,  int timeout) {
+                EXPECT_EQ(string(methodName), _T("Device.X_RDKCENTRAL-COM_T2.UploadDCMReport"));
+                return RBUS_ERROR_SUCCESS;
+            }))
+        .WillOnce(::testing::Invoke(
+            [&](rbusHandle_t handle, char const* methodName, rbusObject_t inParams, rbusMethodAsyncRespHandler_t callback,  int timeout) {
+                return RBUS_ERROR_BUS_ERROR;
+            }));
+
+    ON_CALL(*p_rBusApiImplMock, rbus_close(::testing::_))
+        .WillByDefault(
+            ::testing::Return(RBUS_ERROR_SUCCESS));
+
+    IARM_Bus_PWRMgr_EventData_t param;
+    param.data.state.curState = IARM_BUS_PWRMGR_POWERSTATE_ON;
+    param.data.state.newState = IARM_BUS_PWRMGR_POWERSTATE_STANDBY_LIGHT_SLEEP;
+    powerEventHandler(IARM_BUS_PWRMGR_NAME, IARM_BUS_PWRMGR_EVENT_MODECHANGED, &param, 0);
+
+    status = InvokeServiceMethod("org.rdk.Telemetry.1", "uploadReport", params, result);
+    EXPECT_EQ(Core::ERROR_RPC_CALL_FAILED, status);
+
+    signalled = WaitForRequestStatus(JSON_TIMEOUT,TELEMETRYL2TEST_ONREPORTUPLOAD);
+    EXPECT_TRUE(signalled & TELEMETRYL2TEST_ONREPORTUPLOAD);
+
+    /* Unregister for events. */
+    jsonrpc.Unsubscribe(JSON_TIMEOUT, _T("onReportUpload"));
+}
+
+/********************************************************
+************ PROFILESTATUS USE CASE SCENARIO **************************
+*******************************************************/
+
+TEST_F(Telemetry_L2test, Reportprofilestatus_test){
```

Comment:
Maintain consistency with the test name. Example: TelemetrysetReportProfileStatus (Match with the sequence diagram)